unified-codes/apps/data-loader/src/app/v2/DataParser.ts

```diff
@@ -162,441 +156,87 @@ export class DataParser {
-            };
-
-            this.graph[uc8] = node;
-
-            console.log(`INFO: Created node of type ${type}: ${JSON.stringify(node)}`);
-          }
-        }
-
-        // If row includes unit of presentation code...
-        if (uc7) {
-          const code = uc7;
-          const name = unit_of_presentation;
-          const type = EEntityType.Unit;
-
-          // and node does not exist...
-          if (!(uc7 in this.graph)) {
-            // create unit node.
-            const node = {
-              code,
-              name,
-              type,
-              children: [],
-              properties: [],
-            };
-
-            this.graph[uc7] = node;
-
-            console.log(`INFO: Created node of type ${type}: ${JSON.stringify(node)}`);
-          }
-        }
-
-        // If row include strength code...
-        if (uc6) {
-          const code = uc6;
-          const name = strength;
-          const type = EEntityType.DoseStrength;
-
-          // and node does not exist...
-          if (!(uc6 in this.graph)) {
-            // create strength node.
-            const node = {
-              code,
-              name,
-              type,
-              children: [],
-              properties: [],
-            };
-
-            this.graph[uc6] = node;
-
-            console.log(`INFO: Created node of type ${type}: ${JSON.stringify(node)}`);
-          }
-          // and node exists...
-          else {
+        const productDefinition = [
+          { name: row.product, code: row.uc2, type: EEntityType.Product },
+          { name: row.route, code: row.uc3, type: EEntityType.Route },
+          { name: row.dose_form, code: row.uc4, type: EEntityType.Form },
+          { name: row.dose_qualification, code: row.uc5, type: EEntityType.FormQualifier },
+          { name: row.strength, code: row.uc6, type: EEntityType.DoseStrength },
+          { name: row.unit_of_presentation, code: row.uc7, type: EEntityType.Unit },
+          { name: row.immediate_packaging, code: row.uc8, type: EEntityType.PackImmediate },
+          { name: row.pack_size, code: row.uc9, type: EEntityType.PackSize },
+          // { name: row.outer_packaging, code: row.uc10, type: EEntityType.PackOuter },
+          // { name: row.manufacturer, code: row.uc11, type: EEntityType.Manufacturer },
+          // { name: row.brand, code: row.uc12, type: EEntityType.Brand },
+        ];
+
+        const { uc1, combination } = row;
+
+        const productProperties: IPropertyNode[] = [
+          { type: EPropertyType.RxNav, value: row.rxnav },
+          { type: EPropertyType.WHOEML, value: row.who_eml_product },
+          { type: EPropertyType.NZULM, value: row.nzulm },
+          { type: EPropertyType.UNSPSC, value: row.unspsc }
+        ];
+
+        const itemProperties: IPropertyNode[] = [
+          { type: EPropertyType.WHOEML, value: row.who_eml_item },
+          { type: EPropertyType.NZULM, value: row.who_eml_item },
```

Comment:
good ol' copy paste error :)
```suggestion
          { type: EPropertyType.NZULM, value: row.nzulm_item },
```

Reply:
Ooops, good catch haha 🙀